### DISCUSSION TITLE
Question on setting git lfs download address

### DISCUSSION THREAD

**Discussion by winphuthy** [git-lfs/git-lfs#5566] [Q&A] [answered]
Hey,

Be short, I need to set the git lfs download address in client side, however, I tried to set `remote.{repo}.lfsurl` and `lfs.url`, but it still download large files from origin address.

My git remote repository in my personal gitlab in my own ubuntu pc which has been shared by two internal net, one is wired, two is zerotier internal which is virtual. so there are two address to visit my repo, {ip1} is address in wired internal, {ip2} is address in virtual net. In gitlab, If I set external_url as {ip1} the client in virtual net can reach the git repo but lfs server, if I set external_url as {ip2} all good, but all client in wired internal net will need to join in the virtual net and lose the high speed internal net (ps. we are working on over 100G repo, so high speed in internal net is necessary).

For me the idea is set different download address for the client. But if my plan will not work or some one have more idea to solve this problem, please tell me.

**Comment by chrisd8088** [comment]
Hey, I'm sorry you're having trouble.  I would have expected `lfs.url` and `remote.<remote>.lfsurl` to help, but it's a little difficult to provide more advice without additional information.

You might try using `GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1` with your Git commands to see what URLs are being used by the Git LFS client and what the responses are from the API.

I'd also suggest providing the output of `git lfs env` and `git config -l` (with any private information redacted!!) if you'd like more advice from the community.

**Reply by winphuthy** [reply]
![图片](https://github.com/git-lfs/git-lfs/assets/66393784/efbeccda-d2b1-4818-b909-575042f668e7)
![图片](https://github.com/git-lfs/git-lfs/assets/66393784/f9cb8d31-3029-4260-b5a4-7ecb25375b65)
![图片](https://github.com/git-lfs/git-lfs/assets/66393784/139ef292-0094-40ff-8e7c-a15c4d75993d)
![图片](https://github.com/git-lfs/git-lfs/assets/66393784/884bb291-8811-4b31-84c8-fdf396f98c71)
`GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1` is way too long to paste in here, but the URL its show is the 192.*** ip address.

**Reply by chrisd8088** [reply, marked as answer]
I see the `dial tcp 192.168.0.111:8888` message in one of your screenshots, which looks like it's the hostname in the URLs provided for the individual Git LFS objects.

If you need to rewrite that to something else, like one of your `10.*.*.*` addresses, you could try setting `lfs.transfer.enableHrefRewrite` to `true` and then defining an `insteadOf` Git configuration, perhaps something like:
```
$ git config lfs.transfer.enableHrefRewrite true
$ git config url.http://10.109.235.65:8888/.insteadOf http://192.168.0.111:8888/
```

That should create Git config settings like:
```
[lfs "transfer"]
        enableHrefRewrite = true
[url "http://10.109.235.65:8888/"]
        insteadOf = http://192.168.0.111:8888/
```

As [described](https://github.com/git-lfs/git-lfs/blob/df9b79dc4b96547127c91e03688a7d9cb1619ae7/docs/man/git-lfs-config.adoc#L208-L213) in the manual page, that should rewrite individual Git LFS object URLs which match `http://192.168.0.111:8888/` to instead start with `http://10.109.235.65:8888/`.  You'd want to adjust that to whatever the necessary values are for your actual case, of course.

**Reply by winphuthy** [reply]
Great, It fix my problem, thanks!!!
